connectivity/tests/encryption.go

```diff
@@ -54,53 +75,110 @@ func getInterNodeIface(ctx context.Context, t *check.Test, clientHost *check.Pod
+func getFilter(ctx context.Context, t *check.Test, client, clientHost *check.Pod,
+	server, serverHost *check.Pod,
+	ipFam features.IPFamily, reqType requestType) string {
+
+	tunnelEnabled := false
+	tunnelMode := ""
+	if tunnelStatus, ok := t.Context().Feature(features.Tunnel); ok && tunnelStatus.Enabled {
+		tunnelEnabled = true
+		tunnelMode = tunnelStatus.Mode
 	}
 
-	cmd := []string{
-		"/bin/sh", "-c",
-		fmt.Sprintf("ip -o route get %s | grep -oE 'src [^ ]*' | cut -d' ' -f2",
-			dstIP),
+	protoFilter := ""
+	switch reqType {
+	case requestHTTP:
+		protoFilter = "tcp"
+	case requestICMPEcho:
+		protoFilter = "icmp"
+		if ipFam == features.IPFamilyV6 {
+			protoFilter = "icmp6"
+		}
+	default:
+		t.Fatalf("Invalid request type: %d", reqType)
 	}
-	t.Debugf("Running %s", strings.Join(cmd, " "))
-	srcIP, err := clientHost.K8sClient.ExecInPod(ctx, clientHost.Pod.Namespace,
-		clientHost.Pod.Name, "", cmd)
-	if err != nil {
-		t.Fatalf("Failed to get IP route: %s", err)
+
+	if enc, ok := t.Context().Feature(features.EncryptionPod); tunnelEnabled && ok &&
+		enc.Enabled && enc.Mode == "wireguard" {
+
+		if tunnelMode != "vxlan" {
+			t.Fatalf("Testing %s tunnel mode w/ WireGuard is not supported", tunnelMode)
+		}
+
+		// Captures the following:
+		// - Any VXLAN pkt client host <-> server host. Such a pkt might contain
+		//   a leaked pod-to-pod unencrypted pkt. We could have made this filter
+		//   more fine-grained (i.e., to filter an inner pkt), but that requires
+		//   nasty filtering based on UDP offsets (pcap doesn't support VXLAN).
+		// - Any pkt client <-> server with a given proto. This might be useful
+		//   to catch any regression in the DP which makes the pkt to bypass
+		//   the VXLAN tunnel.
+		filter := fmt.Sprintf("(udp[8:2] = 0x0800 and host %s and host %s) or (host %s and host %s and %s)",
+			clientHost.Address(features.IPFamilyV4), serverHost.Address(features.IPFamilyV4),
+			client.Address(ipFam), server.Address(ipFam), protoFilter)
+
+		return filter
+
 	}
 
-	srcIPStr := strings.TrimRight(srcIP.String(), "\n\r")
-	if srcIPStr != client.Address(ipFam) {
-		filter = fmt.Sprintf("( %s or src host %s )", filter, srcIPStr)
+	filter := fmt.Sprintf("src host %s", client.Address(ipFam))
+	dstIP := server.Address(ipFam)
+
+	if tunnelStatus, ok := t.Context().Feature(features.Tunnel); ok && tunnelStatus.Enabled {
+		cmd := []string{
+			"/bin/sh", "-c",
+			fmt.Sprintf("ip -o route get %s | grep -oE 'src [^ ]*' | cut -d' ' -f2",
+				dstIP),
+		}
+		t.Debugf("Running %s", strings.Join(cmd, " "))
+		srcIP, err := clientHost.K8sClient.ExecInPod(ctx, clientHost.Pod.Namespace,
+			clientHost.Pod.Name, "", cmd)
+		if err != nil {
+			t.Fatalf("Failed to get IP route: %s", err)
+		}
+
+		srcIPStr := strings.TrimRight(srcIP.String(), "\n\r")
+		if srcIPStr != client.Address(ipFam) {
+			filter = fmt.Sprintf("( %s or src host %s )", filter, srcIPStr)
+		}
 	}
+
+	if protoFilter != "" {
```

Comment:
Nit. The `if` check could be dropped, as always guaranteed to be true, and the string concatenation merged with the one below.